Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -2344,6 +2355,35 @@ AODProducerWorkflowDPL::TrackExtraInfo AODProducerWorkflowDPL::processBarrelTrac
   return extraInfoHolder;
 }
 
+bool AODProducerWorkflowDPL::propagateTrackToPV(o2::track::TrackParametrizationWithError<float>& trackPar,
+                                                const o2::globaltracking::RecoContainer& data,
+                                                int colID)
+{
+  if (trackPar.getX() > mMinPropR)
+    return false;
+  o2::dataformats::DCA dcaInfo;
+  dcaInfo.set(999.f, 999.f, 999.f, 999.f, 999.f);
+  o2::dataformats::VertexBase v;
+  if (colID >= 0) {
+    const auto& vtx = data.getPrimaryVertex(colID);
+    v.setPos({vtx.getXYZ()});
+    if (mPropTracksCov) {
+      v.setCov({vtx.getCov()});
+    }
+  } else {
+    v.setPos({mVtx->getXYZ()});
+    if (mPropTracksCov) {
+      v.setCov(mVtx->getSigmaX() * mVtx->getSigmaX(),
+               0.f,
+               mVtx->getSigmaY() * mVtx->getSigmaY(),
+               0.f,
+               0.f,
+               mVtx->getSigmaZ() * mVtx->getSigmaZ());
+    }
+  }
+  return o2::base::Propagator::Instance()->propagateToDCABxByBz(v, trackPar, 2.f, mMatCorr, &dcaInfo);
```

Comment:
Not sure what is the point of `mPropTracksCov` option? You are always propagating the cov.matrix.